python/amici/sbml_import.py

```diff
@@ -1025,14 +1025,13 @@ def _process_rules(self) -> None:
 
             if variable in rulevars:
                 for nested_rule in rules:
-                    nested_formula = sp.sympify(
-                        sbml.formulaToL3String(nested_rule.getMath()),
-                        locals=self.local_symbols)
-                    nested_formula = \
-                        str(nested_formula.subs(variable, formula)).replace('**', '^')
-                    if nested_rule.setFormula(nested_formula) != sbml.LIBSBML_OPERATION_SUCCESS:
-                        raise SBMLException(f'Formula {nested_formula} cannot be parsed by libSBML!')
+                    nested_rule_math_ml = sp.printing.print_mathml(
+                        sbml.formulaToL3String(nested_rule.getMath()))
```

Comment:
Sorry, I'm not sure what `sp.printing.print_mathml` *exactly* does. However, I think the following things are to be considered. If those are fulfilled, things should be fine:
 * replacement of `**` with `^`
 * accounting for local parameters
And I would definitely check the SBML test suite :)
Additionally, if AMICI now can parse additional SBML models, I would account for that in the test-suite, such that we won't support for those any more. :)

Reply:
`print_mathml` generates a mathml string that can be correctly parsed by libsbml, s we don't have to perform any acrobatics here to make libsbml understand what we are outputting here.